tutorials/cloud-functions-async/index.md

```diff
@@ -88,12 +89,15 @@ the task, it moves the job from `Created` to `Running`. When the task is complet
 
 ## Setup
 
-1.  Create a project in the [GCP Console][console].
-1.  [Enable billing for your project.](https://cloud.google.com/billing/docs/how-to/modify-project)
-1.  Use [Cloud Shell][shell] or install the [Google Cloud SDK][sdk].
-1.  Enable Cloud Functions:
+1. Create a project in the [GCP Console][console].
+1. [Enable billing for your project](https://cloud.google.com/billing/docs/how-to/modify-project).
+1. Use [Cloud Shell][shell] or install the [Google Cloud SDK][sdk].
+1. Enable Cloud Functions, Firestore, and Pub/Sub APIs:
```

Comment:
FWIW - I believe pubsub is enabled by default in project, and Firestore gets enabled through still mandatory region select step

Reply:
My quick test shows pubsub isn't enabled by default. From a pedological perspective, not having these here can lead users down a weird path.

> Why did I have to enable the functions API, but not pubsub or firestore? What other APIs do I need to think about...

There's also value in enabling them all at once. gcloud will call EnableServices with the list of all services when given a list, which is one HTTP call and one watch on the batch operation id (the irony in that gcloud and API enablement basically does what your tutorial describes is not lost on me). By separating them out, we end up doing more work, potentially making the user wait longer since the APIs aren't enabled in parallel.

Reply:
yeah - no objection to this change at all - as you say, things change (so I may be wrong now about pubsub, where in the past I'm sure it was enabled by default).